engine/texc/src/texc.cpp

```diff
@@ -65,84 +65,99 @@ namespace dmTexc
     }
 
-    uint32_t GetDataSizeCompressed(HTexture texture, uint32_t mip_map)
+    uint32_t GetDataSizeCompressed(Texture* texture, uint32_t mip_map)
     {
-        Texture* t = (Texture*) texture;
-        return mip_map < t->m_Mips.Size() ? t->m_Mips[mip_map].m_ByteSize : 0;
+        TextureImpl* t = (TextureImpl*)texture->m_Impl;
+        return mip_map < t->m_Mips.Size() ? t->m_Mips[mip_map].m_DataCount : 0;
     }
 
-    uint32_t GetDataSizeUncompressed(HTexture texture, uint32_t mip_map)
+    uint32_t GetDataSizeUncompressed(Texture* texture, uint32_t mip_map)
     {
         // Since we don't add any extra compression on top of the resource, the
         // amount of memory required to store the
         return GetDataSizeCompressed(texture, mip_map);
     }
 
-    uint32_t GetTotalDataSize(HTexture texture)
+    uint32_t GetTotalDataSize(Texture* texture)
     {
-        Texture* t = (Texture*) texture;
+        TextureImpl* t = (TextureImpl*)texture->m_Impl;
         return t->m_Encoder.m_FnGetTotalDataSize(t);
     }
 
-    uint32_t GetData(HTexture texture, void* out_data, uint32_t out_data_size)
+    uint32_t GetData(Texture* texture, void* out_data, uint32_t out_data_size)
     {
-        Texture* t = (Texture*) texture;
+        TextureImpl* t = (TextureImpl*)texture->m_Impl;
         return t->m_Encoder.m_FnGetData(t, out_data, out_data_size);
     }
 
-    uint64_t GetCompressionFlags(HTexture texture)
+    uint64_t GetCompressionFlags(Texture* texture)
     {
-        Texture* t = (Texture*) texture;
+        TextureImpl* t = (TextureImpl*)texture->m_Impl;
         return t->m_CompressionFlags;
     }
 
-    bool Resize(HTexture texture, uint32_t width, uint32_t height)
+    bool Resize(Texture* texture, uint32_t width, uint32_t height)
     {
-        Texture* t = (Texture*) texture;
+        TextureImpl* t = (TextureImpl*)texture->m_Impl;
         return t->m_Encoder.m_FnResize(t, width, height);
     }
 
-    bool PreMultiplyAlpha(HTexture texture)
+    bool PreMultiplyAlpha(Texture* texture)
     {
-        Texture* t = (Texture*) texture;
+        TextureImpl* t = (TextureImpl*)texture->m_Impl;
         return t->m_Encoder.m_FnPreMultiplyAlpha(t);
     }
 
-    bool GenMipMaps(HTexture texture)
+    bool GenMipMaps(Texture* texture)
     {
-        Texture* t = (Texture*) texture;
+        TextureImpl* t = (TextureImpl*)texture->m_Impl;
         return t->m_Encoder.m_FnGenMipMaps(t);
     }
 
-    bool Flip(HTexture texture, FlipAxis flip_axis)
+    bool Flip(Texture* texture, FlipAxis flip_axis)
     {
         if (flip_axis == FLIP_AXIS_Z)
         {
             dmLogWarning("FLIP_AXIS_Z not supported");
             return true;
         }
-        Texture* t = (Texture*) texture;
+        TextureImpl* t = (TextureImpl*)texture->m_Impl;
         return t->m_Encoder.m_FnFlip(t, flip_axis);
     }
 
+    // bool Dither(HTexture texture, PixelFormat pixel_format)
```

Comment:
junk?

Reply:
Kind of a TODO.
It's part of the next PR (that you've now taken over)